### ISSUE TITLE
will this project be updated?

### ISSUE THREAD

**Issue by Zaitonn** [Jither/Jint.DevToolsProtocol#1]
i'm interested in this excellent project but it seems that this project has been out of date so far......
and i've made some effort to support latest Jint but no vain

so .. will you update this recently? or are there any alternatives with devtools support?

**Comment by Zaitonn** [comment]
here are some problems i can't solve
- `Jint.Engine.Parsed` event has been removed
  - i don't know whether `DebugHandler.BeforeEvaluate` is suitable for it 
- some classes are not accessible ('xxxx' is inaccessible due to its protection level)
   ![image](https://github.com/Jither/Jint.DevToolsProtocol/assets/103164490/0572b692-0a88-45b2-8ff9-2e494447258b)

here is my fork: https://github.com/Zaitonn/Jint.DevToolsProtocol

**Comment by Jither** [comment]
To answer the question in the issue title - it's not likely I'll update it. 🙂 I deemed DebugAdapter (i.e. VSCode etc.) generally more useful for debugging (https://github.com/Jither/Jint.DebugAdapter).

`BeforeEvaluate` is indeed intended as the actual implementation of `Parsed` (or in other words, it was the eventual name and signature of `Parsed`) - handling both modules and scripts (hence, it takes a `Program` AST rather than `Script`). But implementing DTP based on that event instead will require some other changes, since the event payload only passes the AST, not the actual source. There are examples of how it's used in both Jint.DebugAdapter and https://github.com/Jither/Jint.DebuggerExample .

As for the internal JsMap, JsSet, JsProxy and (now) JsSymbol, I've posed the question to Marko on Jint - those should probably be returned to public in Jint for consistency with all the other native JS types that are public (JsArray, JsRegExp, JsTypedArray, JsBigInt etc.). The API of Jint has been reduced in preparation for the v3 release, based on public examples of usage, and I guess Map, Set and Proxy were made internal because nothing else uses them, Jint.DevToolsProcotol isn't active, and Jint.DebugAdapter doesn't specifically use those types (they're not considered "special" in DebugAdapter protocol, since - unlike DTP - that protocol isn't specifically based on Javascript).
